lib/rucio/rse/rsemanager.py

```diff
@@ -460,13 +460,17 @@ def upload(rse_settings, lfns, source_dir=None, force_pfn=None, force_scheme=Non
 
                 valid = None
                 try:  # Get metadata of file to verify if upload was successful
-                    stats = protocol.stat(pfn)
-                    if ('adler32' in stats) and ('adler32' in lfn):
-                        valid = stats['adler32'] == lfn['adler32']
-                    if (valid is None) and ('filesize' in stats) and ('filesize' in lfn):
-                        valid = stats['filesize'] == lfn['filesize']
-                except NotImplementedError:
-                    valid = True  # If the protocol doesn't support stat of a file, we agreed on assuming that the file was uploaded without error
+                    try:
```

Comment:
can't you just pull this exception block one level up?

Reply:
The reason I did this nesting is that I wanted to re-raise the exception when `verify_checksum == True` so the standard exception handling can take over. Of, course it is possible to copy the code from there into the `else` part of that if statement.